packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -82,30 +91,32 @@ export interface Store {
 }
 
 export class MemoryStore implements Store {
+  private backend: DBBackend = new MemoryBackend();
   readonly chain: Chain;
-  private _channels: Record<string, MemoryChannelStoreEntry | undefined> = {};
-  private _objectives: Objective[] = [];
-  private _nonces: Record<string, BigNumber | undefined> = {};
   private _eventEmitter = new EventEmitter<InternalEvents>();
-  private _privateKeys: Record<string, string | undefined> = {};
-  private _ledgers: Record<string, string | undefined> = {};
 
-  constructor(privateKeys?: string[], chain?: Chain) {
+  constructor(privateKeys?: string[], chain?: Chain, backend?: DBBackend) {
     // TODO: We shouldn't default to a fake chain
     // but I didn't feel like updating all the constructor calls
     this.chain = chain || new FakeChain();
     this.chain.initialize();
 
+    if (backend) {
+      this.backend = backend;
+    } else {
+      this.backend = new MemoryBackend();
+    }
+
     if (privateKeys && privateKeys.length > 0) {
       // load existing keys
       privateKeys.forEach(key => {
         const wallet = new Wallet(key);
-        this._privateKeys[wallet.address] = wallet.privateKey;
+        this.backend.setPrivateKey(wallet.address, wallet.privateKey);
```

Comment:
Will this cause issues due to `setPrivateKey` being `async` ? If we construct the `store` and then immediately attempt to do something that requires the `privateKey` the `privateKey` might not be set in the `backend` yet.

Reply:
fixed!